Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+          work/inconsistency.
+        - The webdev ones are just in a GH repo. If you don't like 'em,
+          submit a PR.
+        - Style: consistency within a function, then within a file,
+          then within a module, then within a project. Try not to be
+          adaptable. You can always do your favorite style on your
+          personal stuff.
+        - Remember it costs something like $100/hr to have you sit
+          there, and throttle accordingly.
+    - Sooner is better
+        - Even if you aren't comprehensive
+            - You can say “I'm not going to have the time to look at
+              the rest of this for a week. Someone else can feel free
+              to step in.” Don't feel like you married the patch.
+        - Stuff falls out of working memory pretty fast, and it's
+          expensive to load back in.
+        - It's energizing to get your patches turned around quickly.
+- Emotional issues
+    - Insecurity
+          a kind of fear
+        - Who in the room has ever secretly felt (or feared) they're
+          not as competent as the engineers around you?
+              (If they don't put up their hands: "Of course you don't.
+              It's a secret.")
+        - Almost everybody does. It doesn't really go away. Realize
+          nobody's judging you; they're too busy worrying about people
+          judging them.
+        - Realize you've already made it.
+            - If you're hired, congratulations. We don't have an easy
+              hiring process. You're competent to do this job.
+            - Sometimes somebody will find a problem in your PR because
+              they know something you don't. Sometimes it's just having
+              a fresh set of eyes.
+        - Who's afraid their imposter syndrome isn't as good as
+          everybody else's?
+        - It's been a help to me to notice when I'm feeling defensive
+          or angry.
+            - Why am I feeling that way? Sometimes that reveals things
+              I believe that aren't true or things I need to think
+              about.
+            - I generally make bad decisions and say things I regret
+              when feeling that way.
+            - It's a good time to talk a break and come back with a
+              clear head.
+    - Feeling short on time
+          There will always be more to review than can be reviewed. So
+          we have to learn to adapt to that reality: lower standards,
+          never sleep, or just go at a sustainable pace. It can do a
+          lot for a contrib's peace of mind just to know you've seen
+          that the patch is there and that you haven't forgotten about
+          it. Sometimes I even let people know "It's next in my queue."
+        - Consider: what's the risk if there's something wrong with it?
+            - Will it cause a lot of damage?
+            - Will it be easy to fix, or is there, say, a data-exchange
+              format being defined?
```

Comment:
Risk assessment is the cornerstone of how we figure out how much time and energy to spend on things. If it doesn't matter, land it and move on. If it matters a lot, you clear your week and do the review meticulously.

Sometimes I explicitly list what I thought about and looked for in the review in the conclusion figuring I never look at everything, but sometimes it's important to the coder to know what I did look at so they can say, "yo--you need to look at this, too" or so that when we land it and it blows up, we know what we should fix process wise next time around.